starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,141 @@
-// Names and Input
 
+//Create a variable with drivers name
+var hacker1 = "Neil";
 
-//Conditionals
 
 
-// Lorem ipsum generator
+//Print out drivers name
+console.log("The driver's name is: " + hacker1);
+
+
+
+//Ask for Navigators name
+var hacker2 = prompt("Please enter navigator's name:");
+
+
+
+//Print the Navigators name
+console.log("The navigator's name is: " + hacker2);
+
+
+
+//Print response depending on who has longer name
+if (hacker1.length === hacker2.length)
+{
+  console.log("Wow, you both got equally long names, " + hacker1.length + " characters!");
+}
+
+
+
+else if (hacker1.length > hacker2.length)
+{
+  console.log("The driver has the longest name, it has " + hacker1.length + " characters");
+}
+
+
+
+else
+{
+  console.log("Yo, navigator got the longest name, it has " + hacker2.length + " characters.");
+}
+
+
+
+//Print all the characters of the driver's name, separated by a space and in capitals ie. "J O H N"
+
+var s = "";
+var t = "";
+
+for (i = 0; i < hacker1.length; i++)
+{
+  s += hacker1[i].toUpperCase() + " ";
+  
```

Comment:
This will work although we can do the same using JS methods:
```
for (var i = 0; i < 1; i++) {
  console.log(
    hacker1
      .split("")
      .join(" ")
      .toUpperCase()
  );
}
```